### ISSUE TITLE
[Replay] Odd behavior when using  wildcards with OR to search

### ISSUE THREAD

**Issue by Fwang36** [getsentry/sentry#80481]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

Reported by a user but I can reproduce in my org.  Here is a screen recording.  I have several replays where the url comes from localhost.  In my screen recording i search `url: *local*` which returns results then I add a `*host*` in the same condition which then returns no results even though they all come from localhost.  

Using an actual OR in the search bar does work.  So `url:*local* OR url:*host*` works

[Link to my orgs replays](https://francisorg.sentry.io/replays/?project=-1&query=url%3A%5Bhttp%3A%2F%2Flocalhost%3A3000%2F%2Chttp%3A%2F%2Flocalhost%3A3000%2Fabout%5D&statsPeriod=14d)

### Expected Result

Results to be returned even with the wildcards.  Adding multiple full urls with no wildcards does work.

### Actual Result

No results.

https://github.com/user-attachments/assets/0c0d3abd-e6b9-4637-af7c-b8f6776dd7bd

### Product Area

Replays

**Comment by frmsaul** [comment]
Thank you!

**Comment by malwilley** [comment]
> This affects more than Replay it seems, I can repro on Issues' search as well. cc [@malwilley](https://github.com/malwilley) as maybe the new search ui has made this bug more apparent - is this something we want to build support in the backend for?
> 
> The query looks something like this: `query: 'user.email:[*foo*,*bar*]'`. Non wildcards seem to work fine.

You're right, it looks like (for issues search at least) multiple wildcards is not supported ([code here](https://github.com/getsentry/sentry/issues/80678) - looks like we only handle wildcards for single values).

I do think the backend should support this. Created a separate ticket to track on the issues side: https://github.com/getsentry/sentry/issues/80678

**Comment by billyvg** [comment]
Closing this in favor of https://github.com/getsentry/sentry/issues/80678
